### PR TITLE
Basic travis setup

### PR DESCRIPTION
**Pull request by ashcrow**
- Use ``dep`` for dependency management
- Add ``Makefile`` for simple testing/linting/etc..
- Add travis file to test PRs

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -0,0 +1,20 @@
...
+.PHONY: help deps lint test
+
+help:
+	@echo "Targets:"
+	@echo " - deps: Install required tool and dependencies for building"
+	@echo " - lint: Run golint"
+	@echo " - test: Run unittests"
+
+deps:
+	go get -u github.com/golang/dep/cmd/dep
```
Comment (lucab) [inline Makefile]: For this and the one below, I don't like that it messes with user's binaries. What about one of either:
 * moving dep + golint fetch+setup to a dedicated target
 * make `go get` conditional only if the tool is not already available locally

File: Gopkg.lock [hunk 2]
```diff
@@ -0,0 +1,15 @@
+# This file is autogenerated, do not edit; changes may be undone by the next 'dep ensure'.
```
Comment (lucab) [inline Gopkg.lock]: As this is a library-only repo, it shouldn't contain a lockfile.

File: .travis.yml [hunk 3]
```diff
@@ -0,0 +1,23 @@
+language: go
+
+go:
+  - 1.9
```
Comment (lucab) [inline .travis.yml]: Nit: `"1.9.x"` (string + x-version)

Also, any reason why not 1.10 too? Is 1.9 the minimum compat?

**Comment by lucab** [comment]
@ashcrow this needs somebody to log into travis interface and enable this repo there.

**Comment by ashcrow** [comment]
@lucab updated.

@cgwalters mind enabling travis?

**Comment by cgwalters** [comment]
I think it's enabled now.  That said...I'd say we should use PAPR too.

**Comment by ashcrow** [comment]
@cgwalters I'm all for the use of PAPR as well. Can we do that in another PR?

**Comment by ashcrow** [comment]
Bleh, I'm having a heck of a hard time getting ``libostree-dev`` installed in travis. I'll try a few more things. If I can't get it working shortly I'll punt and just go straight to ``papr``.

**Comment by cgwalters** [comment]
It works for me to:

```
# podman run --rm -ti docker.io/ubuntu
Trying to pull docker.io/ubuntu...Getting image source signatures
Copying blob sha256:6b98dfc1607190243b0938e62c5ba2b7daedf2c56d7825dfb835208344705641
 29.81 MB / 29.81 MB [======================================================] 1s
Copying blob sha256:4001a1209541c37465e524db0b9bb20744ceb319e8303ebec3259fc8317e2dec
 843 B / 843 B [============================================================] 0s
Copying blob sha256:6319fc68c576d6bd3e469b0ae31e9a010bc9b71ed286cf4e632424d82dca70d8
 469 B / 469 B [============================================================] 0s
Copying blob sha256:b24603670dc3e91d00439b5d56701884e4fc313877ef3940183a7e8e25fa5f1b
 853 B / 853 B [============================================================] 0s
Copying blob sha256:97f170c87c6f10548068b35cbe9bd00da1278b92f700a5e66c6a16ec04ba456b
 162 B / 162 B [============================================================] 0s
Copying config sha256:113a43faa1382a7404681f1b9af2f0d70b182c569aab71db497e33fa59ed87e6
 3.53 KB / 3.53 KB [========================================================] 0s
Writing manifest to image destination
Storing signatures
root@6c1689c81036:/# apt update
Get:1 http://archive.ubuntu.com/ubuntu bionic InRelease [242 kB]
Get:2 http://security.ubuntu.com/ubuntu bionic-security InRelease [83.2 kB]
Get:3 http://security.ubuntu.com/ubuntu bionic-security/universe Sources [5646 B]
Get:4 http://archive.ubuntu.com/ubuntu bionic-updates InRelease [83.2 kB]       
Get:5 http://security.ubuntu.com/ubuntu bionic-security/multiverse amd64 Packages [1075 B]                
Get:6 http://security.ubuntu.com/ubuntu bionic-security/universe amd64 Packages [38.5 kB]
Get:7 http://archive.ubuntu.com/ubuntu bionic-backports InRelease [74.6 kB]                               
Get:8 http://security.ubuntu.com/ubuntu bionic-security/main amd64 Packages [125 kB]         
Get:9 http://archive.ubuntu.com/ubuntu bionic/universe Sources [11.5 MB]                               
Get:10 http://archive.ubuntu.com/ubuntu bionic/universe amd64 Packages [11.3 MB]           
Get:11 http://archive.ubuntu.com/ubuntu bionic/restricted amd64 Packages [13.5 kB]
Get:12 http://archive.ubuntu.com/ubuntu bionic/multiverse amd64 Packages [186 kB]
Get:13 http://archive.ubuntu.com/ubuntu bionic/main amd64 Packages [1344 kB]
Get:14 http://archive.ubuntu.com/ubuntu bionic-updates/universe Sources [32.7 kB]
Get:15 http://archive.ubuntu.com/ubuntu bionic-updates/main amd64 Packages [186 kB]
Get:16 http://archive.ubuntu.com/ubuntu bionic-updates/multiverse amd64 Packages [1663 B]
Get:17 http://archive.ubuntu.com/ubuntu bionic-updates/universe amd64 Packages [106 kB]
Get:18 http://archive.ubuntu.com/ubuntu bionic-backports/universe amd64 Packages [2793 B]
Fetched 25.4 MB in 4s (6821 kB/s)                         
Reading package lists... Done
Building dependency tree       
Reading state information... Done
8 packages can be upgraded. Run 'apt list --upgradable' to see them.
root@6c1689c81036:/# apt install libostree-dev
Reading package lists... Done
Building dependency tree       
Reading state information... Done
The following additional packages will be installed:
  binutils binutils-common binutils-x86-64-linux-gnu build-essential cpp cpp-7 dconf-gsettings-backend dconf-service dirmngr dpkg-dev fakeroot file g++ g++-7 gcc gcc-7 gcc-7-base gir1.2-glib-2.0
  gir1.2-ostree-1.0 glib-networking glib-networking-common glib-networking-services gnupg gnupg-l10n gnupg-utils gpg gpg-agent gpg-wks-client gpg-wks-server gpgconf gpgsm gpgv gsettings-desktop-schemas
  krb5-locales libalgorithm-diff-perl libalgorithm-diff-xs-perl libalgorithm-merge-perl libarchive13 libasan4 libasn1-8-heimdal libassuan0 libatomic1 libbinutils libc-dev-bin libc6-dev libcc1-0 libcilkrts5
  libdconf1 libdpkg-perl libelf1 libexpat1 libfakeroot libfile-fcntllock-perl libgcc-7-dev libgdbm-compat4 libgdbm5 libgirepository-1.0-1 libglib2.0-0 libglib2.0-bin libglib2.0-data libglib2.0-dev
  libglib2.0-dev-bin libgomp1 libgpgme11 libgssapi-krb5-2 libgssapi3-heimdal libhcrypto4-heimdal libheimbase1-heimdal libheimntlm0-heimdal libhx509-5-heimdal libicu60 libisl19 libitm1 libk5crypto3 libkeyutils1
  libkrb5-26-heimdal libkrb5-3 libkrb5support0 libksba8 libldap-2.4-2 libldap-common liblocale-gettext-perl liblsan0 liblzo2-2 libmagic-mgc libmagic1 libmpc3 libmpdec2 libmpfr6 libmpx2 libnpth0 libostree-1-1
  libpcre16-3 libpcre3-dev libpcre32-3 libpcrecpp0v5 libperl5.26 libproxy1v5 libpython3-stdlib libpython3.6-minimal libpython3.6-stdlib libquadmath0 libreadline7 libroken18-heimdal libsasl2-2 libsasl2-modules
  libsasl2-modules-db libsoup2.4-1 libsqlite3-0 libssl1.1 libstdc++-7-dev libtsan0 libubsan0 libwind0-heimdal libxml2 linux-libc-dev make manpages manpages-dev mime-support multiarch-support netbase patch perl
  perl-base perl-modules-5.26 pinentry-curses pkg-config python3 python3-distutils python3-lib2to3 python3-minimal python3.6 python3.6-minimal readline-common shared-mime-info xdg-user-dirs xz-utils zlib1g-dev
Suggested packages:
  binutils-doc cpp-doc gcc-7-locales dbus-user-session libpam-systemd pinentry-gnome3 tor debian-keyring g++-multilib g++-7-multilib gcc-7-doc libstdc++6-7-dbg gcc-multilib autoconf automake libtool flex bison
  gdb gcc-doc gcc-7-multilib libgcc1-dbg libgomp1-dbg libitm1-dbg libatomic1-dbg libasan4-dbg liblsan0-dbg libtsan0-dbg libubsan0-dbg libcilkrts5-dbg libmpx2-dbg libquadmath0-dbg parcimonie xloadimage scdaemon
  lrzip glibc-doc git bzr gdbm-l10n libglib2.0-doc krb5-doc krb5-user libostree-doc libsasl2-modules-gssapi-mit | libsasl2-modules-gssapi-heimdal libsasl2-modules-ldap libsasl2-modules-otp libsasl2-modules-sql
  libstdc++-7-doc make-doc man-browser ed diffutils-doc perl-doc libterm-readline-gnu-perl | libterm-readline-perl-perl pinentry-doc python3-doc python3-tk python3-venv python3.6-venv python3.6-doc
  binfmt-support readline-doc
The following NEW packages will be installed:
  binutils binutils-common binutils-x86-64-linux-gnu build-essential cpp cpp-7 dconf-gsettings-backend dconf-service dirmngr dpkg-dev fakeroot file g++ g++-7 gcc gcc-7 gcc-7-base gir1.2-glib-2.0
  gir1.2-ostree-1.0 glib-networking glib-networking-common glib-networking-services gnupg gnupg-l10n gnupg-utils gpg gpg-agent gpg-wks-client gpg-wks-server gpgconf gpgsm gsettings-desktop-schemas krb5-locales
  libalgorithm-diff-perl libalgorithm-diff-xs-perl libalgorithm-merge-perl libarchive13 libasan4 libasn1-8-heimdal libassuan0 libatomic1 libbinutils libc-dev-bin libc6-dev libcc1-0 libcilkrts5 libdconf1
  libdpkg-perl libelf1 libexpat1 libfakeroot libfile-fcntllock-perl libgcc-7-dev libgdbm-compat4 libgdbm5 libgirepository-1.0-1 libglib2.0-0 libglib2.0-bin libglib2.0-data libglib2.0-dev libglib2.0-dev-bin
  libgomp1 libgpgme11 libgssapi-krb5-2 libgssapi3-heimdal libhcrypto4-heimdal libheimbase1-heimdal libheimntlm0-heimdal libhx509-5-heimdal libicu60 libisl19 libitm1 libk5crypto3 libkeyutils1 libkrb5-26-heimdal
  libkrb5-3 libkrb5support0 libksba8 libldap-2.4-2 libldap-common liblocale-gettext-perl liblsan0 liblzo2-2 libmagic-mgc libmagic1 libmpc3 libmpdec2 libmpfr6 libmpx2 libnpth0 libostree-1-1 libostree-dev
  libpcre16-3 libpcre3-dev libpcre32-3 libpcrecpp0v5 libperl5.26 libproxy1v5 libpython3-stdlib libpython3.6-minimal libpython3.6-stdlib libquadmath0 libreadline7 libroken18-heimdal libsasl2-2 libsasl2-modules
  libsasl2-modules-db libsoup2.4-1 libsqlite3-0 libssl1.1 libstdc++-7-dev libtsan0 libubsan0 libwind0-heimdal libxml2 linux-libc-dev make manpages manpages-dev mime-support multiarch-support netbase patch perl
  perl-modules-5.26 pinentry-curses pkg-config python3 python3-distutils python3-lib2to3 python3-minimal python3.6 python3.6-minimal readline-common shared-mime-info xdg-user-dirs xz-utils zlib1g-dev
The following packages will be upgraded:
  gpgv perl-base
2 upgraded, 138 newly installed, 0 to remove and 6 not upgraded.
Need to get 71.5 MB of archives.
After this operation, 314 MB of additional disk space will be used.
Do you want to continue? [Y/n] 
```

Is the travis `go` language ubuntu version older?  It may be easier to just `apt install golang libostree-dev` in a "base" setup or so?

**Comment by ashcrow** [comment]
FWIW I was able to get things working in a container. However ``github.com/ostreedev/ostree-go/pkg/otbuiltin``'s test always fails. This happens on my local host as well as in a container.

**Comment by ashcrow** [comment]
@cgwalters with ``bionic`` it will work as the packages are there. However, ``trusty`` is what Travis uses. And if we are going to use a container I'll probably opt for Fedora :-).

**Comment by ashcrow** [comment]
The current test here works as long as we disable unittesting. As noted, I was unable to get one of the tests to pass no matter where I ran the test.

File: .travis.yml [hunk 4]
```diff
@@ -1,28 +1,22 @@
...
   email: false
-
     
 before_script:
-  - wget http://mirrors.kernel.org/ubuntu/pool/universe/o/ostree/libostree-dev_2018.4-2_amd64.deb
-  - sudo dpkg -i --force-all ./libostree-dev_2018.4-2_amd64.deb
-  - mkdir -p $HOME/gopath/src/github.com/ostreedev/ostree-go/
-  - rsync -az ${TRAVIS_BUILD_DIR}/ $HOME/gopath/src/github.com/ostreedev/ostree-go/
+  - docker pull registry.fedoraproject.org/fedora:latest
+  - docker run -e GOPATH=/opt --cidfile=/tmp/cidfile registry.fedoraproject.org/fedora:latest /bin/bash -c "dnf install -y make golang go-compilers-golang-compiler findutils ostree ostree-devel glib-devel glib"
```
Comment (cgwalters) [inline .travis.yml]: You want `glib2`, not `glib` (though neither should be explicitly necessary anyways, ostree will pull it in via deps).

**Comment by lucab** [comment]
@ashcrow if you are not in a hurry for having the unittest, I can tackle that later as anyway I have to rebase and check my #14.

**Pull request merged**